src/ex2/script.js

```diff
@@ -0,0 +1,146 @@
+class Main {
+    constructor() { }
+
+    init() {
+        //add click event listener to add button
+        addBtn.addEventListener("click", () => { this.addTodoAndRender() })
+
+        // add a task when user press Enter key
+        taskInput.addEventListener("keypress", (e) => { if (e.key === 'Enter' && taskInput.value) this.addTodoAndRender() });
+
+        //Clear the tasks list
+        clearAllBtn.addEventListener("click", () => {
+            itemManager.clearArr();
+            taskList.innerHTML = ''; //Clear the list of tasks element
+            this.updateForm();
```

Comment:
makes more sense for the render function that decides if to render the tasks or not by length of the array and then updates form and not decide in each case what to do and what not here, this is more big prone would be better if there is one place that is responsible for these decisions